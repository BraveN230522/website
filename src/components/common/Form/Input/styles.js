import styled from '@emotion/styled';

const InputGroup = styled.div`
  width: 22rem;
`;

const InputComponent = styled.input`
  ${({ icon }) => icon && `background: url(${icon}) no-repeat scroll 16px center`};
  border: 2px solid ${({ theme, light }) => (light ? theme.colors.nobel : theme.colors.white)};
  border-radius: 25px;
  font-family: ${({ theme }) => theme.fonts.Inter};
  color: ${({ theme, light }) => (light ? theme.colors.dark : theme.colors.white)};
  display: block;
  font-size: 13px;
  outline: none;
  font-weight: 400;
  width: 100%;
  line-height: 20px;
  padding-left: ${({ light }) => (light ? '2rem' : '2.5rem')};
  padding-right: 0.8rem;
  font-weight: 400;
  height: 45px;

  ::placeholder {
    color: ${({ theme, light }) => (light ? theme.colors.suvaGrey : theme.colors.white)};
    opacity: 1;
  }

  ${({ theme }) => theme.media.phone`
    padding: 0 2.4rem;
  `};
`;

const MessageContainer = styled.div`
  margin-top: 5px;
`;

const AdviceMessage = styled.span`
  color: ${({ theme }) => theme.colors.greyishbrown};
  font-weight: 600;
  display: block;
  font-size: 12px;
  padding-left: 20px;
`;

const InputLabel = styled.label`
  font-size: 18px;
  line-height: 24px;
  letter-spacing: -0.12px;
  margin-left: 9px;
`;

export { InputGroup, InputLabel, InputComponent, MessageContainer, AdviceMessage };
