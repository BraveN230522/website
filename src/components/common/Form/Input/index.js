import React, { forwardRef } from 'react';

import Message from '../Message';

import { InputGroup, InputComponent, MessageContainer, InputLabel, AdviceMessage } from './styles';

const Input = forwardRef(
  (
    {
      advice,
      className,
      error,
      label,
      icon,
      id,
      placeholder,
      value,
      name,
      onChange,
      onFocus,
      type = 'text',
      onBlur = () => null,
      light,
    },
    ref
  ) => (
    <InputGroup className={className}>
      {Boolean(label) && <InputLabel htmlFor={id}>{label}</InputLabel>}
      <InputComponent
        light={light}
        className={className}
        icon={icon}
        name={name}
        ref={ref}
        type={type}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        placeholder={placeholder}
        value={value}
      />
      {!!error && <Message type="error">{error}</Message>}
      {Boolean(advice) && (
        <MessageContainer>
          <AdviceMessage>{advice}</AdviceMessage>
        </MessageContainer>
      )}
    </InputGroup>
  )
);

export default Input;
