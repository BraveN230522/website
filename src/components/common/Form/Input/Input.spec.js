import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Input from './index';

/**
 * TODO: When graphql request is configured,
 * this toHaveContent tests will be changed for a mock response
 */
describe('<Input />', () => {
  it('Renders the Input unchanged', () => {
    const { container } = render(<Input />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
