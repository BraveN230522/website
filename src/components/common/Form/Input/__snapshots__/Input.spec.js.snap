// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renders the Input unchanged 1`] = `
.emotion-2 {
  width: 22rem;
}

.emotion-0 {
  border: 2px solid #ffffff;
  border-radius: 25px;
  font-family: Inter,sans-serif;
  color: #ffffff;
  display: block;
  font-size: 13px;
  outline: none;
  font-weight: 400;
  width: 100%;
  line-height: 20px;
  padding-left: 2.5rem;
  padding-right: 0.8rem;
  font-weight: 400;
  height: 45px;
}

.emotion-0::-webkit-input-placeholder {
  color: #ffffff;
  opacity: 1;
}

.emotion-0::-moz-placeholder {
  color: #ffffff;
  opacity: 1;
}

.emotion-0:-ms-input-placeholder {
  color: #ffffff;
  opacity: 1;
}

.emotion-0::placeholder {
  color: #ffffff;
  opacity: 1;
}

@media (max-width:48em) {
  .emotion-0 {
    padding: 0 2.4rem;
  }
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <input
      class="emotion-0 emotion-1"
      value=""
    />
  </div>
</div>
`;
